qiskit_ibm_runtime/noise_learner.py

```diff
@@ -0,0 +1,276 @@
+                    validate_isa_circuits([task], self._backend.target)
+
+                if isinstance(self._backend, IBMBackend):
+                    self._backend.check_faulty(task)
+
+        logger.info("Submitting job using options %s", learner_options)
+
+        # Batch or Session
+        if self._mode:
+            return self._mode.run(
+                program_id=self._program_id(),
+                inputs=inputs,
+                options=runtime_options,
+                callback=options_dict.get("environment", {}).get("callback", None),
+                result_decoder=DEFAULT_DECODERS.get(self._program_id()),
+            )
+
+        if self._backend:
+            runtime_options["backend"] = self._backend
+            if "instance" not in runtime_options and isinstance(self._backend, IBMBackend):
+                runtime_options["instance"] = self._backend._instance
+
+        if isinstance(self._service, QiskitRuntimeService):
+            return self._service.run(
+                program_id=self._program_id(),
+                options=runtime_options,
+                inputs=inputs,
+                callback=options_dict.get("environment", {}).get("callback", None),
+                result_decoder=DEFAULT_DECODERS.get(self._program_id()),
+            )
+
+        return self._service.run(
+            program_id=self._program_id(),  # type: ignore[arg-type]
+            options=runtime_options,
+            inputs=inputs,
+        )
+
+    @classmethod
+    def _program_id(cls) -> str:
+        """Return the program ID."""
+        return "noise-learner"
+
+    def _set_options(
+        self, options: Optional[Union[Dict, NoiseLearnerOptions, EstimatorOptions]] = None
+    ) -> None:
+        """
+        Sets the options, ensuring that they are of type ``NoiseLearnerOptions``.
+        """
+        if not options:
+            self._options = NoiseLearnerOptions()
+        elif isinstance(options, NoiseLearnerOptions):
+            self._options = options
+        elif isinstance(options, EstimatorOptions):
+            d = asdict(options.resilience.layer_noise_learning)  # type: ignore[union-attr]
+            d.update({"twirling_strategy": options.twirling.strategy})  # type: ignore[union-attr]
+            d.update({"max_execution_time": options.max_execution_time})
+            self._options = NoiseLearnerOptions(**d)
+        else:
+            self._options = NoiseLearnerOptions(**options)
+
+    @staticmethod
+    def _get_inputs_options(options_dict: dict[str, Any]) -> dict[str, str]:
+        """Returns a dictionary of options that must be included in the program inputs,
+        filtering out every option that is not part of the NoiseLearningOptions."""
+        ret = {}
+
+        for key in [
+            "max_layers_to_learn",
+            "shots_per_randomization",
+            "num_randomizations",
+            "layer_pair_depths",
+            "twirling_strategy",
+            "simulator",
```

Comment:
I'd suggest using the `dataclasses.fields()` method to extract all fields from `NoiseLearnerOptions` then remove the ones you don't want. When things change, it's a lot easier to find an extra parameter (since it'd be rejected by the server), then one that's set and not passed. 

Also I assume you meant to include `experimental`?